### PR TITLE
LocationAwareSlf4jLogger does not correctly format log message.

### PR DESCRIPTION
**Pull request by normanmaurer**
Motivation:

We did miss to use MessageFormatter inside LocationAwareSlf4jLogger and so {} was not correctly replaced in log messages when using slf4j.
This regression was introduced by afe0767e9c150369cd6ea150b5cc1a9021fd5c3c.

Modifications:

- Make use of MessageFormatter
- Add unit test.

Result:

Fixes https://github.com/netty/netty/issues/8483.

### CODE REVIEW

**Comment by normanmaurer** [comment]
@franz1981 @njhill PTAL as well if you have time

File: common/src/main/java/io/netty/util/internal/logging/LocationAwareSlf4JLogger.java [hunk 1]
```diff
@@ -54,28 +54,28 @@ public boolean isTraceEnabled() {
...
-            log(TRACE_INT, msg, null);
+            log(TRACE_INT, msg);
         }
     }
 
     @Override
     public void trace(String format, Object arg) {
         if (isTraceEnabled()) {
-            log(TRACE_INT, format, arg);
+            log(TRACE_INT, org.slf4j.helpers.MessageFormatter.format(format, arg).getMessage(), arg);
```
Comment (njhill) [inline common/src/main/java/io/netty/util/internal/logging/LocationAwareSlf4JLogger.java]: Presumably there's no need to also pass the args now? If so it would be better to pass null for the last arg to avoid varargs array allocation.

Comment (normanmaurer) [reply common/src/main/java/io/netty/util/internal/logging/LocationAwareSlf4JLogger.java]: from other implementations I think you still should pass the args.

Comment (njhill) [reply common/src/main/java/io/netty/util/internal/logging/LocationAwareSlf4JLogger.java]: I still don't follow (sorry I'm probably missing something) ... the additional args are only passed for the purpose of later substitution into the message, but now the substitution is done upfront. So the message passed won't have `{}` corresponding to the args..?

Comment (normanmaurer) [reply common/src/main/java/io/netty/util/internal/logging/LocationAwareSlf4JLogger.java]: @njhill when looking at other implementation they all forwarded the "args", so I did the same as I thought maybe they are used somehow later on. I think at worse you have one more alloc but not passing these down may produce other side-effects depending on the underlying implementations.

That said maybe you are right and we should just not do it.

Comment (normanmaurer) [reply common/src/main/java/io/netty/util/internal/logging/LocationAwareSlf4JLogger.java]: @njhill PTAL again... I think this addresses your concern and is also cleaner :)

Comment (njhill) [reply common/src/main/java/io/netty/util/internal/logging/LocationAwareSlf4JLogger.java]: @normanmaurer LGTM, I was about to suggest something similar :)

File: common/src/main/java/io/netty/util/internal/logging/LocationAwareSlf4JLogger.java [hunk 2]
```diff
@@ -54,28 +58,28 @@ public boolean isTraceEnabled() {
     @Override
     public void trace(String msg) {
         if (isTraceEnabled()) {
```
Comment (njhill) [inline common/src/main/java/io/netty/util/internal/logging/LocationAwareSlf4JLogger.java]: Not part of this PR but I think methods like this one don't need the `if (isXxxEnabled())` right?

**Comment by 91he** [comment]
There is still a bug. 

When you use "Object... argArray" as an arg, you shall use "MessageFormatter.arrayFormat".

**Comment by njhill** [comment]
@91he I think you are right ... want to open a PR?

**Comment by njhill** [comment]
@91he I've opened a PR #8834 to fix, thanks for reporting this!
